web/api/v1/api.go

```diff
@@ -274,10 +295,16 @@ func (api *API) query(r *http.Request) (interface{}, *apiError, func()) {
 		qs = stats.NewQueryStats(qry.Stats())
 	}
 
+	var warnings []error
+	if fanout, ok := newQueryable.(storage.FanoutStorage); ok {
+		warnings = fanout.GetRemoteErrors()
```

Comment:
Warnings should probably be part of the main interface

Reply:
The warnings are passed to the main interface, and then removed from queryData later. I did this to try and avoid editing the apiFunc interface. Would it be preferable to add a new value to apiFunc? ie. `type apiFunc func(r *http.Request) (interface{}, *apiError, []error, func())` where the []error is the warnings returned? It would remove the type checking at line 179 then too.

Reply:
That sounds like a better approach. We should probably change that to a struct while we're at it, 4 return values is pushing it.